DNN Platform/Library/Services/Mail/CoreMailProvider.cs

```diff
@@ -161,32 +161,21 @@ public override string SendMail(MailInfo mailInfo, SmtpInfo smtpInfo = null)
                             case "1": // basic
                                 if (!string.IsNullOrEmpty(smtpInfo.Username) && !string.IsNullOrEmpty(smtpInfo.Password))
                                 {
-                                    smtpClient.UseDefaultCredentials = false;
-                                    smtpClient.Credentials = new NetworkCredential(smtpInfo.Username, smtpInfo.Password);
+                                    smtpClient.Authenticate(smtpInfo.Username, smtpInfo.Password);
                                 }
 
                                 break;
-                            case "2": // NTLM
-                                smtpClient.UseDefaultCredentials = true;
+                            case "2": // NTLM (Not Supported by MailKit)
```

Comment:
This is a breaking change technically.  Are we ok with this?  I think so but calling attention to it.

Reply:
Yes. We'll just have to document it.

> You cannot use CredentialCache.DefaultNetworkCredentials with MailKit because when MailKit asks the credential to provide the username and password strings that it needs to send to the server, the DefaultNetworkCredentials tells MailKit that the username and password strings are empty and so MailKit tries to authenticate with empty strings.
> 
> System.Net.Mail seems to be able to use an internal API to get the real strings that MailKit does not have access to.

Here's the reference thread I found: https://stackoverflow.com/questions/38113437/is-it-possible-to-use-default-network-credentials-with-mailkit-and-exchange